deps/undici/src/lib/observable.js

```diff
@@ -0,0 +1,802 @@
+          },
+        },
+        {
+          signal: destination.signal,
+        }
+      );
+    });
+  }
+
+  drop(count) {
+    return new Observable((destination) => {
+      let seen = 0;
+      this.subscribe(
+        {
+          next(value) {
+            seen++;
+            if (seen > count) {
+              destination.next(value);
+            }
+          },
+          error(error) {
+            destination.error(error);
+          },
+          complete() {
+            destination.complete();
+          },
+        },
+        {
+          signal: destination.signal,
+        }
+      );
+    });
+  }
+
+  flatMap(project) {
+    return new Observable((destination) => {
+      let queue = [];
+      let index = 0;
+      let innerAC;
+      let outerComplete = false;
+      const startInner = (value) => {
+        innerAC = new AbortController();
+        const signal = AbortSignal.all([innerAC.signal, destination.signal]);
+        let innerObservable;
+        try {
+          innerObservable = Observable.from(project(value, index++));
+        } catch (error) {
+          destination.error(error);
+          return;
+        }
+        innerObservable.subscribe(
+          {
+            next(innerValue) {
+              destination.next(innerValue);
+            },
+            error(error) {
+              destination.error(error);
+            },
+            complete() {
+              innerAC = undefined;
+              if (queue.length > 0) {
+                startInner(queue.shift());
+              } else if (outerComplete) {
+                destination.complete();
+              }
+            },
+          },
+          {
+            signal,
+          }
+        );
+      };
+
+      this.subscribe(
+        {
+          next(value) {
+            if (innerAC) {
+              queue.push(value);
+            } else {
+              startInner(value);
+            }
+          },
+          error(error) {
+            destination.error(error);
+          },
+          complete() {
+            outerComplete = true;
+            if (queue.length === 0) {
+              destination.complete();
+            }
+          },
+        },
+        {
+          signal: destination.signal,
+        }
+      );
+    });
+  }
+
+  takeUntil(notifier) {
+    return new Observable((destination) => {
+      Observable.from(notifier).subscribe(
+        {
+          next() {
+            destination.complete();
+          },
+          error(error) {
+            destination.error(error);
+          },
+        },
+        {
+          signal: destination.signal,
+        }
+      );
+      this.subscribe(destination, { signal: destination.signal });
+    });
+  }
+
+  every(predicate, options) {
+    return new Promise((resolve, reject) => {
+      const ac = new AbortController();
+      const signal = options?.signal
+        ? AbortSignal.all([ac.signal, options.signal])
+        : ac.signal;
```

Comment:
Should be more performant to de-reference "signal" once:

```suggestion
      const signal0 = options?.signal;
      const signal = signal0
        ? AbortSignal.all([ac.signal, signal0])
        : ac.signal;
```